src/averageAges.js

```diff
@@ -15,11 +15,24 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const peopleFilter
+    = people.filter(person => person.sex === 'm' && filterCentury(person));
+
+  function filterCentury(person) {
+    if (century !== undefined) {
+      const age = Math.ceil(person.died / 100);
+
+      return age === century;
+    }
+
+    return true;
+  }
+
+  const reopleAgeArray
+    = peopleFilter.map(person => person.died - person.born);
+
+  return reopleAgeArray.reduce((a, b) => a + b)
+  / reopleAgeArray.length;
```

Comment:
you can combine these methods.

```suggestion
  return peopleFilter.reduce(
    (sumOfAges, person) => sumOfAges + (person.died - person.born),
    0,
  ) / reopleAgeArray.length;
```
`- 1 loop )`

but better create for reduce result separate variable

Reply:
Also, you do the same on calculateWomenAverageAge. You can create a separate function for it and use it here and on calculateWomenAverageAge